pypsa/plot.py

```diff
@@ -247,27 +256,40 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
         bus_collection = ax.scatter(x, y, c=c, s=s, cmap=bus_cmap, edgecolor='face')
 
     def as_branch_series(ser):
+        # ensure that this function always return a multiindexed series
         if isinstance(ser, dict) and set(ser).issubset(branch_components):
-            return pd.Series(ser)
-        elif isinstance(ser, pd.Series):
-            if isinstance(ser.index, pd.MultiIndex):
-                return ser
-            index = ser.index
-            ser = ser.values
+            return network.branches().loc[branch_components] \
+                          .join(pd.Series(ser, name='value') \
+                          .rename_axis('component')).value
+        elif isinstance(ser, pd.Series) and isinstance(ser.index, pd.MultiIndex):
+                return ser.rename_axis(['component', 'name'])
```

Comment:
Use `rename(index= ...)` or `rename(columns=...)`, instead of `rename_axis`. It's clearer and you can change both at the same time.

Reply:
`rename(index=...)` is only changing values in the index, not the name of the index axis, which is intended here. So I have to stick to this one. but at least now it is `.rename_axis(index=['component', 'name'])`